modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp

```diff
@@ -266,7 +266,13 @@ boolean showRestoreButton = (searchTerms.getStatus() != WorkflowConstants.STATUS
 		}
 		else {
 			if (filterManageableOrganizations && !UserPermissionUtil.contains(permissionChecker, ResourceConstants.PRIMKEY_DNE, ActionKeys.VIEW)) {
-				userParams.put("usersOrgs", ArrayUtil.toLongArray(user.getOrganizationIds()));
+				long[] organizationIds = user.getOrganizationIds();
+
+				if (ArrayUtil.isEmpty(organizationIds)) {
+					organizationIds = new long[] {0};
```

Comment:
Hey Jorge,

user.getOrganizationIds() returns an empty string if the user does not belong to any organization, in which i think the implementation is correct. The problem is when using the `usersOrgs` param, it does a joint/filter on the users_orgs table on the search results. If the `usersOrgs` param has an empty array, the results wont be filtered based on the organizations. Because of that, a user without any organization is able to view all the users. We need to add a default organization id when organizationIds is empty to ensure that the results is filtered correctly.